storage/src/lib.rs

```diff
@@ -127,7 +133,7 @@ macro_rules! try_open {
 
 impl Storage {
     pub fn init(cfg: &StorageConfig) -> Result<Self> {
-        let mut lookups = BTreeMap::new();
+        let mut lookups = LinkedHashMap::with_capacity(1000);
```

Comment:
1000 is completely arbitrary here. `LinkedHashMap` implements auto-re-alloc on insert if out of capacity, so this one only ensures there will not be many re-allocations on the initial read. We have 73 epochs per year atm, so 1000 is way ahead of limitations. Maybe we can read the current number of existing indexes here first and init the map with this capacity plus some buffer.